EIPS/eip-5283.md

```diff
@@ -0,0 +1,107 @@
+---
+eip: 5283
+title: Semaphore for Parallelizable Reentrancy Guard
+description: A Precompile-based parallelizable reentrancy protection using the call stack
+author: Sergio D. Lerner (@SergioDemianLerner)>
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-07-17
+---
```

Comment:
```suggestion
requires: 20, 1109, 1153, 1283, 1352, 2046
---
```

Reply:
Should an EIP referenced be in the "requires" list even if it doesn't need to be final for this EIP to be finalized ? For example, should competing proposals be referenced in "requires" ?

Reply:
You generally shouldn't include competing proposals in your EIP.  An EIP should stand on its own as much as possible, and external references should be to final EIPs (or EIPs that are expected to be final before your EIP makes it to final).  Keep in mind, the EIP process is for standardization and isn't an appropriate place to argue for/against this standard vs that standard.  Such material is better suited for a blog, article, docs site, discussions-to link, forum thread, etc.

My usual recommendation to authors is to remove all references to other EIPs that aren't required reading for understanding this standard, and if you want to mention something in the rationale as "we thought about doing X but decided against it because Y" then just briefly describe X rather than linking to a non-final EIP.

Keep in mind, EIPs can be completely/radically changed prior to becoming final so if you write your EIP and reference a non-final EIP then it is possible that the content you referenced may no longer make sense, which is why we don't allow non-final references in final EIPs.